### DISCUSSION TITLE
How to apply control in viewer?

### DISCUSSION THREAD

**Discussion by andrealu2012** [google-deepmind/mujoco#1692] [Asking for Help]
i am using python in Mujoco.i can write control law in .py and run this .py. but when i run viewer and drag .xml file into viewer,the control law cannot work(because mujoco cannot  connect .py to this .xml file,i guess),so help,how to apply control law in .py in viewer? Thank you

**Comment by aftersomemath** [comment]
You can either use the passive Python viewer or callbacks.

I have two examples of using callbacks here:
https://github.com/aftersomemath/mujoco_python_viewer_examples
